src/Extensions/ServiceFabricGuest.Abstractions/options/ServiceFabricRestClientOptions.cs

```diff
@@ -1,19 +1,52 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT license.
-
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
 using System.ComponentModel.DataAnnotations;
-
+using System.Linq;
+using Microsoft.Omex.Extensions.Abstractions;
+
 namespace Microsoft.Omex.Extensions.ServiceFabricGuest.Abstractions
 {
 	/// <summary>
 	/// Option for the REST communication with cluster
 	/// </summary>
 	public class ServiceFabricRestClientOptions
-	{
+	{
 		/// <summary>
-		/// Settings definition for Cluster Endpoint
+		/// Settings definition for Cluster Endpoint Port
+		/// Value taken from https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-visualizing-your-cluster
+		/// </summary>
+		public int ClusterEndpointPort { get; set; } = 19080;
+
+		/// <summary>
+		/// Settings definition for Cluster Endpoint FQDN
 		/// </summary>
 		[Required(AllowEmptyStrings = false)]
-		public string ClusterEndpoint { get; set; } = string.Empty;
+		public string ClusterEndpointFQDN { get; set; } = DefaultFQDN();
+
+
+		/// <summary>
+		/// Settings definition for Cluster Endpoint 
+		/// </summary>
+		public string ClusterEndpoint => string.Format("http://{0}:{1}", ClusterEndpointFQDN, ClusterEndpointPort.ToString());

```

Comment:
I don't like that we are recalculating value each time.
Ideally, I would prefer value recalculating only after changes in FQDN or port, but I understand it would require a lot of changes to the class, so maybe we could at least replace this property with a method instead to express this behavior better?